### PR TITLE
feat(source-gcs): add support for zip files

### PR DESCRIPTION
**Pull request by darynaishchenko**
## What
resolved: https://github.com/airbytehq/airbyte-internal-issues/issues/8949
could not set up source for zip files

## How
`smart_open`, which source uses for opening files, doesn't support .zip compression. For reading zip files unzipping into a temporary folder was added. Then source uses path to it as uri to remote file. GCSRemoteFile is used to avoid using temp path in state and record values for uri field. 

## Review guide

1. `airbyte-integrations/connectors/source-gcs/source_gcs/stream_reader.py`
2. `airbyte-integrations/connectors/source-gcs/source_gcs/helpers.py`
3. `airbyte-integrations/connectors/source-gcs/source_gcs/cursor.py`
4. `airbyte-integrations/connectors/source-gcs/source_gcs/stream.py`
5. other files

## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by aldogonzalez8 — APPROVED** [review]
LGTM

File: airbyte-integrations/connectors/source-gcs/source_gcs/stream.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    """
+
+    def transform_record(self, record: dict[str, Any], file: GCSRemoteFile) -> dict[str, Any]:
+        file_datetime_string = file.last_modified.strftime(self.DATE_TIME_FORMAT)
+
+        record[self.ab_last_mod_col] = file_datetime_string
+        record[self.ab_file_name_col] = file.displayed_uri if file.displayed_uri else file.uri
+        return record
+
+    def read_records_from_slice(self, stream_slice: StreamSlice) -> Iterable[AirbyteMessage]:
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-gcs/source_gcs/stream.py:35]: I'm not into the idea of copypaste the whole method to change 1 line only.
it feels like we need to refactor this method in CDK to pass custom `transform_record`

**Review by artem1205 — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-gcs/source_gcs/stream.py (hunk 1 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-gcs/source_gcs/stream.py]: created pr for cdk https://github.com/airbytehq/airbyte/pull/45698

**Review by artem1205 — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]
lgtm

**Pull request merged**